src/spark_rapids_ml/clustering.py

```diff
@@ -270,7 +270,13 @@ def _cuml_fit(
                 concated = pd.concat(df_list)
             else:
                 # should be list of np.ndarrays here
-                concated = np.concatenate(df_list)
+                rows = sum([arr.shape[0] for arr in df_list])
+                cols = df_list[0].shape[1]
+                d_type = df_list[0].dtype
+                concated_out = np.empty(shape=(rows, cols), order="F", dtype=d_type)
+                concated = np.concatenate(df_list, out=concated_out)
+                for df in df_list:
+                    del df
```

Comment:
Could we put this into a utils function to be shared for all algos

Reply:
Done.